src/domain/tokens/entities/schemas/token.schema.ts

```diff
@@ -1,37 +1,16 @@
-import { JSONSchemaType, Schema } from 'ajv';
-import { buildPageSchema } from '@/domain/entities/schemas/page.schema.factory';
-import { Token, TokenType } from '@/domain/tokens/entities/token.entity';
+import { buildZodPageSchema } from '@/domain/entities/schemas/page.schema.factory';
+import { TokenType } from '@/domain/tokens/entities/token.entity';
+import { AddressSchema } from '@/validation/entities/schemas/address.schema';
+import { z } from 'zod';
 
-export const TOKEN_SCHEMA_ID =
-  'https://safe-client.safe.global/schemas/tokens/token.json';
+export const TokenSchema = z.object({
+  address: AddressSchema,
+  decimals: z.number().nullish().default(null),
+  logoUri: z.string(),
+  name: z.string(),
+  symbol: z.string(),
+  type: z.nativeEnum(TokenType),
+  trusted: z.boolean(),
+});
 
-export const tokenSchema: JSONSchemaType<Token> = {
-  $id: TOKEN_SCHEMA_ID,
-  type: 'object',
-  properties: {
-    address: { type: 'string' },
-    decimals: { oneOf: [{ type: 'number' }, { type: 'null', nullable: true }] },
-    logoUri: { type: 'string' },
-    name: { type: 'string' },
-    symbol: { type: 'string' },
-    type: { type: 'string', enum: Object.values(TokenType) },
-    trusted: { type: 'boolean' },
-  },
-  required: [
-    'address',
-    'decimals',
-    'logoUri',
-    'name',
-    'symbol',
-    'type',
-    'trusted',
-  ],
-};
-
-export const TOKEN_PAGE_SCHEMA_ID =
-  'https://safe-client.safe.global/schemas/tokens/token-page.json';
-
-export const tokenPageSchema: Schema = buildPageSchema(
-  TOKEN_PAGE_SCHEMA_ID,
-  tokenSchema,
-);
+export const TokenPageSchema = buildZodPageSchema(TokenSchema);
```

Comment:
Can we add tests for these?

Reply:
Added in [316e401](https://github.com/safe-global/safe-client-gateway/pull/1317/commits/316e401981f992491f825cf2420f3cece771a46f)

And adapted to the new constraints in [cccaf2e](https://github.com/safe-global/safe-client-gateway/pull/1317/commits/cccaf2ede0ebd3f067c10490ef7b1270cf250163)